dask_ml/model_selection/_split.py

```diff
@@ -478,12 +511,36 @@ def train_test_split(
                     f" dask versions<2.13.0. Current version is {DASK_VERSION}."
                 )
             kwargs = {}
-        return list(
-            itertools.chain.from_iterable(
+
+        if stratify is not None:
+            train_test_pairs = []
+            for arr in arrays:
+                # list of class-wise train/test split dataframe slices
+                arr_train_slices = []
+                arr_test_slices = []
+                for ci in classes:
+                    # get subdf of data from this class
+                    ci_arr = arr[stratify == ci]
+
+                    # split subdf
+                    arr_train, arr_test = ci_arr.random_split(
+                        [train_size, test_size], random_state=rng, **kwargs,
+                    )
+
+                    # add subdf's from this class to list of all subdf's
+                    arr_train_slices.append(arr_train)
+                    arr_test_slices.append(arr_test)
+
+                # concat all train subdfs as 1 train df, same for test
+                train_test_pairs.append(
+                    [dd.concat(arr_train_slices), dd.concat(arr_test_slices)]
+                )
```

Comment:
For the output is not ordered by classes. I think we needs to add some kind of shuffle here.
```suggestion
                train = dd.concat(arr_train_slices)
                test = dd.concat(arr_test_slices)
                train = train.shuffle(train.index)
                test = test.shuffle(test.index)
                # concat all train subdfs as 1 train df, same for test
                train_test_pairs.append([train, test])
```